app.js

```diff
@@ -1,78 +1,111 @@
+const kaholoPluginLibrary = require("kaholo-plugin-library");
 const Docker = require("dockerode");
 const path = require("path");
 const {
-  getUrl, getAuth, streamFollow, execCmd, isFile,
+  getUrl,
+  mapParamsToAuthConfig,
+  streamFollow,
+  execCmd,
+  isFile,
 } = require("./helpers");
 
 const docker = new Docker();
 
-async function build(action) {
-  const imageTag = action.params.TAG.trim();
-  let inputPath = action.params.PATH.trim();
+async function build({
+  TAG: imageTag,
+  PATH,
+}) {
+  let inputPath = PATH.trim();
   if (!inputPath) {
-    // eslint-disable-next-line no-throw-literal
-    throw "Must provide docker file path";
+    throw new Error("Must provide docker file path");
   }
 
   if (isFile(inputPath)) {
-    inputPath = path.join(inputPath, "..");
+    inputPath = path.dirname(inputPath);
   }
 
   const cmd = `docker build ${imageTag ? `-t ${imageTag} ` : ""}${inputPath}`;
   return execCmd(cmd);
 }
 
-async function pull(action, settings) {
-  const auth = getAuth(action, settings);
-  const imageUrl = getUrl(action.params.URL, action.params.IMAGE, action.params.TAG);
+async function pull({
+  URL: url,
+  IMAGE: image,
+  TAG: imageTag,
+  ...authParams
+}) {
+  const auth = mapParamsToAuthConfig(authParams);
```

Comment:
You're even renaming it to `authConfig` when passing to `docker.pull`. Why not just name it properly at the first place?